src/libsystemd/meson.build

```diff
@@ -82,80 +124,50 @@ sd_login_sources = files('sd-login/sd-login.c')
 ############################################################
 
 libsystemd_sources = files(
-        'sd-bus/bus-common-errors.c',
-        'sd-bus/bus-common-errors.h',
         'sd-bus/bus-container.c',
         'sd-bus/bus-container.h',
         'sd-bus/bus-control.c',
         'sd-bus/bus-control.h',
         'sd-bus/bus-convenience.c',
         'sd-bus/bus-creds.c',
         'sd-bus/bus-creds.h',
-        'sd-bus/bus-dump.c',
-        'sd-bus/bus-dump.h',
         'sd-bus/bus-error.c',
         'sd-bus/bus-error.h',
-        'sd-bus/bus-internal.c',
-        'sd-bus/bus-internal.h',
-        'sd-bus/bus-introspect.c',
-        'sd-bus/bus-introspect.h',
         'sd-bus/bus-kernel.c',
         'sd-bus/bus-kernel.h',
-        'sd-bus/bus-match.c',
-        'sd-bus/bus-match.h',
         'sd-bus/bus-message.c',
         'sd-bus/bus-message.h',
         'sd-bus/bus-objects.c',
         'sd-bus/bus-objects.h',
+        'sd-bus/bus-protocol.c',
         'sd-bus/bus-protocol.h',
-        'sd-bus/bus-signature.c',
-        'sd-bus/bus-signature.h',
         'sd-bus/bus-slot.c',
         'sd-bus/bus-slot.h',
         'sd-bus/bus-socket.c',
         'sd-bus/bus-socket.h',
         'sd-bus/bus-track.c',
         'sd-bus/bus-track.h',
-        'sd-bus/bus-type.c',
-        'sd-bus/bus-type.h',
         'sd-bus/sd-bus.c',
-        'sd-device/device-enumerator-private.h',
         'sd-device/device-enumerator.c',
-        'sd-device/device-internal.h',
-        'sd-device/device-monitor-private.h',
         'sd-device/device-monitor.c',
-        'sd-device/device-private.c',
-        'sd-device/device-private.h',
-        'sd-device/device-util.c',
-        'sd-device/device-util.h',
         'sd-device/sd-device.c',
         'sd-hwdb/hwdb-internal.h',
         'sd-hwdb/sd-hwdb.c',
         'sd-netlink/netlink-genl.c',
         'sd-netlink/netlink-genl.h',
-        'sd-netlink/netlink-internal.h',
-        'sd-netlink/netlink-message-nfnl.c',
         'sd-netlink/netlink-message-rtnl.c',
         'sd-netlink/netlink-message.c',
         'sd-netlink/netlink-slot.c',
         'sd-netlink/netlink-slot.h',
         'sd-netlink/netlink-socket.c',
-        'sd-netlink/netlink-types-genl.c',
-        'sd-netlink/netlink-types-internal.h',
-        'sd-netlink/netlink-types-nfnl.c',
-        'sd-netlink/netlink-types-rtnl.c',
-        'sd-netlink/netlink-types.c',
-        'sd-netlink/netlink-types.h',
```

Comment:
Let me confirm that files in src/libsystemd/internal are build twice for libsystemd and libsystemd-shared, right?
And, files in src/libsystemd except for those in internal are only build for libsystemd?

If my understanding is correct, then these netlink-types*.[ch] should be in libsystemd/sd-netlink, as IIRC, these are not directly used by files in shared or other executables.

Reply:
> Let me confirm that files in src/libsystemd/internal are build twice for libsystemd and libsystemd-shared, right? And, files in src/libsystemd except for those in internal are only build for libsystemd?

Yes.

> If my understanding is correct, then these netlink-types*.[ch] should be in libsystemd/sd-netlink, as IIRC, these are not directly used by files in shared or other executables.

"directly" is not relevant. What matters whether they are used *only* through `_public_` functions, in which case it is enough to link to `libsystemd`, or whether they are used by directly or through another non-public function. In this case, `netlink_seal_message` calls `message_seal`, and `nfnl_message_batch` which calls `message_new` which calls `type_system_root_get_type_system_and_header_size`, and then the dependency tree cascades. So maybe it'd be possible to split this better, but I think it'll require a rethinking of how `netlink-types*.[ch]` are supposed to be used. We can always do that in a later PR.